### ISSUE TITLE
How to import css from lib?

### ISSUE THREAD

**Issue by spredemann** [transitive-bullshit/create-react-library#7]
I'm trying to import a css file from the lib into the example page but couldn't figure it out and I was also faced with a weird cli behavior.

Whenever I use a path that was invalid, the terminal got stuck into Compiling... and opening the Activity Monitor the node process was over 100% cpu somehow. This only happens when trying to import via ` import 'my-component/styles.css'`. Importing the component itself via `import MyComponent from 'my-component'` works fine and importing other css files from the same dir as the example also work fine. Oh and if I try to import some other non existent file from the example dir it doesn't cause the cli to get stuck.

**Comment by transitive-bullshit** [comment]
Interesting. I'm not sure about the compiling issue, as the example is using `create-react-app` directly with no changes.  Maybe because it's locally linked, there's some weird edge case with CRA?

You should never need to import css as a user of your library. Rather, you should import the css within the library itself, which'll get bundled into the library via [rollup-plugin-css](https://github.com/egoist/rollup-plugin-postcss).

[Here](https://github.com/transitive-bullshit/react-modern-library-boilerplate/blob/master/src/index.js#L8) is an example of what I'm talking about.  The library imports the css directly. Rollup bundles that css along with your javascript.  Then, the user of your module (or the example app in this case) just imports your module's JS and the accompanying CSS "just works".

**Comment by spredemann** [comment]
What if my component provides some basic styles (that I'm ok with importing directly) and themes that are optional and want to showcase those in my examples?

**Comment by transitive-bullshit** [comment]
Hmmm, I'm not sure how to bundle css directly.  Maybe try copying the css to `dist/` directly?

One other option would be to include all the possible css with your module and then enable only the styles / theme you want a runtime via props and [classnames](https://github.com/JedWatson/classnames), which is a really helpful utility module I use a lot.

And lastly, if you get stuck, you can always ask the folks over at [rollup](https://rollupjs.org), as I'm sure this has come up before.

**Comment by spredemann** [comment]
Thanks for the help so far but I'm still having trouble. I did look at your example and while it would work fine for the very basic styles, including all themes when they are not going to be used doesn't seem to be the best way to go about it.

The CRA getting stuck in Compiling... definitely seems to be some edge case thing. However I tried moving the css file to `/dist `and I still get `Module not found: Can't resolve`.

Now I'm really confused cause it should have worked? I don't know in depth how linking works but I suppose when I import my lib in the example it's just looking for the package in the `/dist` folder and therefore any files in there should be reachable.

**Comment by transitive-bullshit** [comment]
imho, given that the css will be extremely small I would honestly recommend including all themes unless you literally have thousands of styles.

From an end user's point-of-view, your library will be easier to use by controlling theming via JS as opposed to controlling theming by including a specific css file.  It also allows for dynamic use cases like changing theming on-the-fly.

I can think of very few use cases where I wouldn't personally recommend including all the CSS directly, as the overhead will be negligible compared to the size of basic JS dependencies.

**Comment by spredemann** [comment]
Thanks. I figured it out... I just had to import the css using the full path like `import 'my-component/dist/styles.css'` even though the component itself can be imported like `import MyComponent from 'my-component'`.

I'm still doing more research regarding css modules but I'm not sold yet. I see some big libs using name spaced classes and I've used and customized those with no problem. I get your point in terms of usability but generally the user will have to add more styles and override to make the component match the app anyway.
